plugins/otter-pro/inc/plugins/class-form-pro-features.php

```diff
@@ -512,6 +513,158 @@ public function replace_magic_tags( $content, $form_inputs ) {
 		return $content;
 	}
 
+	/**
+	 * Mark request with Stripe as temp.
+	 *
+	 * @param Form_Data_Request|null $form_data The form data.
+	 */
+	public function mark_request_with_stripe_as_temp( $form_data ) {
+		if ( ! isset( $form_data ) ) {
+			return $form_data;
+		}
+
+		if (
+			( ! class_exists( 'ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request' ) ) ||
+			! ( $form_data instanceof \ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request ) ||
+			$form_data->has_error()
+		) {
+			return $form_data;
+		}
+
+		$fields_options = $form_data->get_wp_fields_options();
+
+		foreach ( $fields_options as $field ) {
+			if ( $field->has_type() && 'stripe' === $field->get_type() ) {
+				$form_data->mark_as_temporary();
+				break;
+			}
+		}
+
+		return $form_data;
+	}
+
+	/**
+	 * Create a Stripe session.
+	 *
+	 * @param Form_Data_Request|null $form_data The form data.
+	 * @see https://stripe.com/docs/api/checkout/sessions/create
+	 */
+	public function create_stripe_session( $form_data ) {
+		if ( ! isset( $form_data ) ) {
+			return $form_data;
+		}
+
+		if (
+			( ! class_exists( 'ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request' ) ) ||
+			! ( $form_data instanceof \ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request ) ||
+			$form_data->has_error() ||
+			$form_data->is_duplicate()
+		) {
+			return $form_data;
+		}
+
+		if ( ! $form_data->has_metadata( 'otter_form_record_id' ) ) {
+			return $form_data;
+		}
+
+		$has_stripe = false;
+
+		$fields_options = $form_data->get_wp_fields_options();
+
+		foreach ( $fields_options as $field ) {
+			if ( $field->has_type() && 'stripe' === $field->get_type() ) {
+				$has_stripe = true;
+				break;
+			}
+		}
+
+		if ( ! $has_stripe ) {
+			return $form_data;
+		}
+
+		$required_fields = $form_data->get_wp_options()->get_required_fields();
+
+		$products_to_process = array();
+
+		foreach ( $fields_options as $field ) {
+			if (
+				$field->has_name() &&
+				'stripe' === $field->get_type() &&
+				in_array( $field->get_name(), $required_fields, true ) &&
+				$field->has_stripe_product_info()
+			) {
+				$products_to_process[] = $field->get_stripe_product_info();
+			}
+		}
+
+		if ( empty( $products_to_process ) ) {
+			return $form_data;
+		}
+
+		$payload = array(
+			'mode' => 'payment',
+		);
+
+		$permalink = add_query_arg(
+			array(
+				'stripe_checkout' => '{CHECKOUT_SESSION_ID}', // Testing mode.
+			),
+			$form_data->get_data_from_payload( 'postUrl' )
+		);
+
+		$payload['success_url'] = $permalink;
+		$payload['cancel_url']  = $permalink;
+
+		// For @HardeepAsrani: by pre-filling the customer email, it will appear in the Stripe Checkout page without the possibility to change it. You can try it first then we can remove it if you want.
```

Comment:
We can remove pre-filled email as you explained it won't let users change email on checkout page.